modules/i3s/src/lib/utils/customize-сolors.ts

```diff
@@ -79,7 +79,18 @@ export async function customizeColors(
     if (!color) {
       continue; // eslint-disable-line no-continue
     }
-    colors.value.set(color, i * 4);
+
+    /* eslint max-statements: ["error", 30] */
+    /* eslint complexity: ["error", 12] */
+    if (options.i3s.colorsByAttribute.mode === 'multiply') {
+      const multupliedColor = colors.value.subarray(i * 4, i * 4 + 3);
```

Comment:
Nit: Generates a lot of small objects, perf / GC.

What happens if you just multiply all elements in the array "in situ"?